### PR TITLE
Update all versions and remove vm_service_client dependency

### PR DESCRIPTION
**Pull request by jonahwilliams**
rolls new versions and migrates the devicelab off of package:vm_service_client due to pub_semver pin

Work towards https://github.com/flutter/flutter/issues/77856

### CODE REVIEW

**Comment by zanderso** [comment]
What was the issue the last time in the device lab for switching to `package:vm_service` and how is that issue avoided this time?

**Comment by jonahwilliams** [comment]
> What was the issue the last time in the device lab for switching to package:vm_service and how is that issue avoided this time?

framework.dart sends an invalid VM service response - the vm service requires a JSON formatted string and we were sending a string. the old client was overly permissive, the new one crashes. I added a work around for that, because I'm a bit concerned also about changing the handshake while flutter/cocoon has its own runner

**Comment by jonahwilliams** [comment]
(That said, there could be other issues :)  )

File: dev/devicelab/lib/framework/runner.dart [hunk 1]
```diff
@@ -160,17 +163,55 @@ Future<VMIsolateRef> _connectToRunnerIsolate(Uri vmServiceUri) async {
...
+}
+
+/// The cocoon client sends an invalid VM service response, we need to intercept it.
+Future<VmService> vmServiceConnectUri(String wsUri, {Log log}) async {
+  final WebSocket socket = await WebSocket.connect(wsUri);
+  final StreamController<dynamic> controller = StreamController<dynamic>();
+  final Completer<dynamic> streamClosedCompleter = Completer<dynamic>();
+  socket.listen(
+    (dynamic data) {
+      final Map<String, dynamic> rawData = json.decode(data as String) as Map<String, dynamic> ;
```
Comment (jonahwilliams) [inline dev/devicelab/lib/framework/runner.dart:199]: This is where the fun begins

**Comment by christopherfujino** [comment]
> I added a work around for that, because I'm a bit concerned also about changing the handshake while flutter/cocoon has its own runner

Do we still use the flutter/cocoon agent?

File: dev/devicelab/lib/framework/runner.dart (hunk 1 above)
Comment (zanderso) [reply dev/devicelab/lib/framework/runner.dart:199]: Consider adding an `onError` callback to the `listen` call that will propagate a useful stack trace.

**Comment by jonahwilliams** [comment]
Yes, it is still used for a few of the windows tasks

**Review by zanderso — COMMENTED** [review]
To make it easier to isolate any problems that pop up, is there any way to handle the package updates and switching device lab to `package:vm_service.dart` separately?

**Comment by jonahwilliams** [comment]
Yeah, that I can do

**Comment by jonahwilliams** [comment]
Updated so this is only the version roll.

**Review by zanderso — APPROVED** [review]
rslgtm

**Comment by jonahwilliams** [comment]
This is a  version update and will not affect g3

**Pull request merged**